modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -336,31 +336,86 @@ PolygonConcentricCircleMeshGeneratorBase::PolygonConcentricCircleMeshGeneratorBa
+        paramError("ring_block_ids",
+                   "This parameter must have the appropriate size if it is provided: "
+                   "If the number of the innermost ring layers is more than one, "
+                   "and we have quad elements, the size of 'ring_block_ids' must be "
+                   "equal to the size of 'ring_intervals' or 'ring_intervals' + 1. Else, equal to "
+                   "the size of 'ring_intervals'\n",
+                   debug_info.str());
+      }
+    }
+    // If conditions are met, duplicate the first element of _ring_block_names at the start.
     if (!_ring_block_names.empty() && _quad_center_elements && num_innermost_ring_layers > 1 &&
         _ring_block_names.size() == _ring_intervals.size())
       _ring_block_names.insert(_ring_block_names.begin(), _ring_block_names.front());
+    // check if the number of ring block names is appropiate
     if (!_ring_block_names.empty() &&
         _ring_block_names.size() !=
             (_ring_intervals.size() + (unsigned int)(num_innermost_ring_layers != 1)))
-      paramError(
-          "ring_block_names",
-          "This parameter must have the appropriate size if it is set. The size should be the "
-          "same as the size of 'ring_intervals' if the innermost ring interval (including "
-          "boundary layers) is unity; otherwise the size should be greater than the size of "
-          "'ring_intervals' by one. If 'quad_center_elements' is true, it is optional to only "
-          "provide this parameter with the same size as 'ring_intervals'");
+    {
+      // Create an ostringstream for the debug information
+      std::ostringstream debug_info;
+      debug_info << "quad_center_elements is : " << (_quad_center_elements ? "true" : "false")
+                 << std::endl;
+      debug_info << "ring_block_names size is : " << _ring_block_names.size() << std::endl;
+      debug_info << "ring_intervals size is : " << _ring_intervals.size() << std::endl;
+      debug_info << "number of innermost ring layers is : " << num_innermost_ring_layers
+                 << std::endl;
+
+      // error message
+      if (!_quad_center_elements)
+      {
+        paramError("ring_block_names",
+                   "This parameter must have the appropriate size if it is provided: "
+                   "If the number of the innermost ring layers is more than one, "
+                   "and we have non-quad elements, the size of 'ring_block_names' must be "
+                   "equal to the size of 'ring_intervals' + 1. Else, equal to the size of "
+                   "'ring_intervals'\n",
+                   debug_info.str());
+      }
+      else
+      {
+        paramError("ring_block_names",
+                   "This parameter must have the appropriate size if it is provided: "
+                   "If the number of the innermost ring layers is more than one, "
+                   "and we have quad elements, the size of 'ring_block_names' must be "
```

Comment:
customize this message more, we know we have all quads in this case

Reply:
we know that we either have or not have quad central elements. If we mention that we don't have quad central elements in one error message we should also mention that we have them in the other.

Reply:
this "we have quad elements" is inside the 'if' condition, so we are not asserting that we have quads here

Reply:
I think this whole thing would be a lot clearer if you had a message structured like this:
- there are this many inner most ring layers
- we are / are not using quad elements for the center elements
Therefore the number of block ids should equal to the number ... ( <that number> )